### DISCUSSION TITLE
Update search params partially by default

### DISCUSSION THREAD

**Discussion by JKupski** [TanStack/router#2107] [Q&A]
I don't like the way I have to update my search parameters when having a search state with more than one property.
`navigate({
            search: (prev) => ({
              ...prev,
              tab: tvalue,
            })
          })`

Is there an idea for a better solution or is it an option to modify the update behaviour of the navigate state, so that by default only the state passed is updated? When I want to set my state to null or undefined, I want to do that explicitly.

What are your thoughts about this behaviour? Better solutions are welcome :)

**Comment by schiller-manuel** [comment]
what exactly don't you like?
